File: python/passphrase.py
#!/usr/local/bin/python3

def testPhrase(phrase):
    wp = []
    for w in phrase.split():
        if w in wp:
            return False
        wp.append(w)
    return True


assert True == testPhrase("aa bb cc dd ee")
assert False == testPhrase("aa bb cc dd aa")
assert True ==  testPhrase("aa bb cc dd aaa")

File: nim/passphrase.nim
import strutils
import sequtils

proc isPhraseValid(p: string): bool =
  var wp: seq[string] = @[]
  for v in p.split:
    if v in wp:
      return false
    wp.add(v)
  return true

assert true == isPhraseValid("aa bb cc dd ee")
assert false == isPhraseValid("aa bb cc dd aa")
assert true ==  isPhraseValid("aa bb cc dd aaa")

var valid=0
for line in lines "../phraseinput.txt":
  if isPhraseValid(line):
    valid += 1
echo valid
